pkg/collect/utilisation.go

```diff
@@ -0,0 +1,366 @@
+package collect
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/mesosphere/dkp-cli-runtime/core/output"
+	coreout "github.com/mesosphere/dkp-cli-runtime/core/output"
+	"github.com/phayes/freeport"
+	"github.com/pkg/errors"
+	prom "github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/httpstream"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+// UsageReading the CPU and memory utilisation at a given timestamp
+type UsageReading struct {
+	Timestamp time.Time `json:"timestamp,omitempty" yaml:"timestamp,omitempty"`
+	CPU       float64   `json:"cpu,omitempty" yaml:"cpu,omitempty"`
+	Memory    float64   `json:"memory,omitempty" yaml:"memory,omitempty"`
+}
```

Comment:
Have you checked if there is a some known data format for storing metrics? Maybe from prometheus itself. It would be great for interoperability and further data processing.